docs/sdkx-xamarin/notifications-android.mdx

```diff
@@ -0,0 +1,226 @@
+[Service(Exported = true)]
+[IntentFilter(new[] {"com.google.firebase.MESSAGING_EVENT"})]
+public class MyFirebaseMessagingService : FirebaseMessagingService
+{
+        const string TAG = "MyFirebaseMsgService";
+
+        public override void OnMessageReceived(RemoteMessage message)
+        {
+            IDictionary<string, string> data = message.Data;
+            string origin = data.ContainsKey("origin") ? data["origin"] : null;
+            if (origin != null && origin.Equals("helpshift"))
+            {
+                // Let Helpshift SDK handle this notification.
+                Helpshift.HandlePush(data, false);
+                return;
+            } else {
+                // Handle other notifications.
+            }
+        }
+}
+```
+
+## In-app notifications {#in-app-notifications}
+
+In-app notifications are similar to notifications in the [notification drawer](http://developer.android.com/guide/topics/ui/notifiers/notifications.html) . Unlike push notifications, they appear only when you app is running.
+
+These notifications are sent when an agent replies to a customer's issue. Your customers can go straight into the [conversation screen](/sdkx_android/support-tools#conversation-view) when they tap on the notification.
+
+<Centered >
+
+![](/static/books/sdkx_android/notification.png)
+
+</Centered>
+
+<Admonition type="info" title="Note">
+
+If the FCM device token is registered for push notifications, then in-app notifications will be disabled. In-app notifications are disabled to avoid duplicate notifications from both push notifications and in-app notifications.
+
+</Admonition>
+
+## Showing notification count when replies are sent to the user {#showing-notification-count}
+
+To fetch unread messages count from the server you can call `Helpshift.RequestUnreadMessageCount(shouldFetchFromServer)` API. This API will return unread messages count via delegate.
+Based on the value of `shouldFetchFromServer`, the locally stored count will be returned if `shouldFetchFromServer` is `false` else from the server by fetching remotely when `shouldFetchFromServer` is `true`. An example use of this count is to update the badge count to indicate unread messsages.
+Please note that before calling this method, you need to set the listener for Helpshift events by calling the `Helpshift.SetHelpshiftEventsListener(eventsListener)` API.
+
+```csharp
+
+namespace HelpshiftXamarinTestApp.Droid
+{
+    [Application]
+    public class MyApplication : Application
+    {
+        public const string CHANNEL_ID = "HelpshiftXamarinDemo";
+
+        public MyApplication(IntPtr handle, JniHandleOwnership transfer) : base(handle, transfer)
+        {
+            // do any initialisation you want here (for example initialising properties)
+        }
+
+        override public void OnCreate()
+        {
+            base.OnCreate();
+
+            Dictionary<String, Object> installC = new Dictionary<String, Object>();
```

Comment:
string and object should begin the smallcase right?

Reply:
no, this works